src/sentry/integrations/gitlab/seach.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+from sentry.api.bases.organization import OrganizationEndpoint, OrganizationPermission
+from sentry.models import Integration
+
+
+class GitlabIssueSearchEndpoint(OrganizationEndpoint):
+    permission_classes = (OrganizationPermission, )
+
+    def get(self, request, organization, integration_id):
+        try:
+            integration = Integration.objects.get(
+                organizations=organization,
+                id=integration_id,
+                provider='gitlab',
```

Comment:
I don't think you need `organization` nor `provider` if you have the `id.

Reply:
Adding additional fields won't cause it to look at any different indexes or anything like that. It'd hit the index used by the primary key, return the 1 row since it's UNIQUE, then it'd apply a filter on the extra bits to that one row.

The reason we do this instead of relying on just simply an id is to guard against potentialyl incorrect data. So scoping to organization_id as well here like we're doing is good to me since it prevents an integration from outside of the org to be selected (if for whtaever reason maybe we had a bug where you'd in whatever scenario get an id for an integration not in this org). So we overqualify some things to make sure they don't leak outside of our bounds.

I'd argue that adding `provider='gitlab'` is probably excessive, but that's your call if you think maybe that'd happen where you'd have an `id` and it's _not_ gitlab, and you want to make sure it is, then this makes sense.

But in terms of database performance, it won't matter. It doesn't check more indexes or anything, it just will apply a filter for `(organization = 1) AND (provider = gitlab )` to the one row returned.

If you're curious or want to understand more, you can run this query through an `EXPLAIN` and see what the query plan is.

Reply:
I guess double checking in this case, `organizations` is a M2M relationship, so it'll  have to do a JOIN on this `OrganizationIntegration` table, in this case, it's not a simple filter, and will touch another table, but I still agree that an extra double check is fine. My point still stands with the provider name though.

Usually passing around simple ids can lead to potential security holes in weird ways if you don't always double check what the id is pointing to. A simple example of this being say, an API endpoint that took an id as a querystrin argument. You can query the id from the databse, but you still don't know if it's an id they're allowed to see, etc.